### DISCUSSION TITLE
Getting the access token, and injecting into all fetch() requests

### DISCUSSION THREAD

**Discussion by simplenotezy** [nextauthjs/next-auth#2206] [Help]
## Question 💬
I'd like to inject the accessToken into all requests called by my helper function, I currently have a helper function like so:

    authedFetcher('/admin/update', { method: 'POST', body: {....} });

But I have to inject the authToken in every request, which is not very pretty. Do we have a way to fetch the accessToken without using `context` or using `useSession()`? (which can only be used inside React components, and not in custom helper functions).

**Comment by mahieyin-rahmun** [comment]
Maybe you can refactor your fetcher into a custom hook that can use the `useSession()` hook.
